packages/ses/src/lockdown-shim.js

```diff
@@ -0,0 +1,35 @@
+// @ts-check
+
+// We import this first to fail as fast as possible if the shim has been
+// transformed or embedded in a way that causes it to run in sloppy mode.
+// See https://github.com/endojs/endo/blob/master/packages/ses/error-codes/SES_NO_SLOPPY.md
+import './assert-sloppy-mode.js';
+import { globalThis } from './commons.js';
+import { repairIntrinsics } from './lockdown.js';
+
+/**
+ * @param {import('./lockdown.js').LockdownOptions} options
+ */
+globalThis.lockdown = options => {
+  const hardenIntrinsics = repairIntrinsics(options);
+  globalThis.harden = hardenIntrinsics();
+};
+
+/**
+ * @param {import('./lockdown.js').LockdownOptions} options
+ */
+globalThis.repairIntrinsics = options => {
+  const hardenIntrinsics = repairIntrinsics(options);
+  // Reveal hardenIntrinsics after repairs.
+  globalThis.hardenIntrinsics = () => {
+    // Reveal harden after hardenIntrinsics.
+    // Harden is dangerous before hardenIntrinsics because hardening just
+    // about anything will inadvertently render intrinsics irreparable.
+    // Also, for modules that must work both before or after lockdown (code
+    // that is portable between JS and SES), the existence of harden in global
+    // scope signals whether such code should attempt to use harden in the
+    // defense of its own API.
+    // @ts-ignore harden not yet recognized on globalThis.
+    globalThis.harden = hardenIntrinsics();
+  };
```

Comment:
I’ve moved all of the `globalThis` assignment behaviors into the shim to decontaminate the non-shim modules.

So, note that the global `repairIntrinsics` doesn’t return `hardenIntrinsics` and the global `hardenIntrinsics` does not return `harden`, but the internal versions perform this cascade.

`hardenIntrinsics` did not previously return `harden`.